### ISSUE TITLE
The trained generator model returns different output each time for a same input image

### ISSUE THREAD

**Issue by kalai2033** [junyanz/pytorch-CycleGAN-and-pix2pix#828]
Hi,
   I have used a pix2pix model to train a model with my custom dataset. I noticed something weird when I tested the generator model, it returns a different output each time for a same input image with slight changes. Is it because of the non deterministic behaviour of GAN or am i doing something wrong here?

**Comment by junyanz** [comment]
It's normal as the default pix2pix generator has Dropout, and by default, the test code still enables Dropout for randomness. If you don't like Dropout, you can disable it by setting `--no_dropout` for both training and test.

**Comment by kalai2033** [comment]
Thanks @junyanz

**Comment by kalai2033** [comment]
I trained a pix2pix model with --no_dropout flag. And i got same output every time when i tested the images. And when i used --eval flag while testing the model trained without --no_dropout flag, i had the same behaviour. So what is difference between --eval and --no_dropout flags?
